subprojects/plugins/src/main/java/org/gradle/api/plugins/internal/SourceSetUtil.java

```diff
@@ -52,13 +53,21 @@ public File call() {
         }));
     }
 
-    public static void configureAnnotationProcessorPath(final SourceSet sourceSet, CompileOptions options, final Project target) {
-        new DslObject(options).getConventionMapping().map("annotationProcessorPath", new Callable<Object>() {
+    public static void configureAnnotationProcessorPath(final SourceSet sourceSet, SourceDirectorySet sourceDirectorySet, CompileOptions options, final Project target) {
+        final ConventionMapping conventionMapping = new DslObject(options).getConventionMapping();
+        conventionMapping.map("annotationProcessorPath", new Callable<Object>() {
             @Override
             public Object call() {
                 return sourceSet.getAnnotationProcessorPath();
             }
         });
+        final String annotationProcessorGeneratedSourcesChildPath = "generated/sources/annotationProcessor/" + sourceDirectorySet.getName() + "/" + sourceSet.getName();
+        conventionMapping.map("annotationProcessorGeneratedSourcesDirectory", new Callable<Object>() {
```

Comment:
Should this use `setAnnotationProcessorGeneratedSourcesDirectory(Provider<File>)` instead of convention mapping?